Formula/k/kubelogin.rb

```diff
@@ -0,0 +1,24 @@
+class Kubelogin < Formula
+  desc "OpenID Connect authentication plugin for kubectl"
+  homepage "https://github.com/int128/kubelogin"
+  url "https://github.com/int128/kubelogin.git",
+    tag:      "v1.28.1",
+    revision: "852a040607b8d7c14dad3db1eb72674f0df63f13"
+  license "Apache-2.0"
+  head "https://github.com/int128/kubelogin.git", branch: "master"
+
+  depends_on "go" => :build
+  depends_on "kubernetes-cli"
+
+  def install
+    ldflags = %W[
+      -s -w
+      -X main.version=v#{version}
+    ]
+    system "go", "build", *std_go_args(output: bin/"kubectl-oidc_login", ldflags:)
+  end
+
+  test do
+    assert_match "v#{version}", shell_output("#{bin}/kubectl-oidc_login --version")
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.